x/ccv/consumer/keeper/distribution.go

```diff
@@ -140,3 +140,37 @@ func (k Keeper) GetConnectionHops(ctx sdk.Context, srcPort, srcChan string) ([]s
 	}
 	return ch.ConnectionHops, nil
 }
+
+// GetEstimatedNextFeeDistribution returns data about next fee distribution. Data represents an estimation of
+// accumulated fees at the current block height.
+func (k Keeper) GetEstimatedNextFeeDistribution(ctx sdk.Context) (types.NextFeeDistributionEstimate, error) {
+	lastH, err := k.GetLastTransmissionBlockHeight(ctx)
+	if err != nil {
+		return types.NextFeeDistributionEstimate{}, err
+	}
+
+	nextH := lastH.GetHeight() + k.GetBlocksPerDistributionTransmission(ctx)
+
+	consumerFeePoolAddr := k.authKeeper.GetModuleAccount(ctx, k.feeCollectorName).GetAddress()
+	total := k.bankKeeper.GetAllBalances(ctx, consumerFeePoolAddr)
+
+	frac, err := sdk.NewDecFromStr(ConsumerRedistributeFrac)
+	if err != nil {
+		return types.NextFeeDistributionEstimate{}, err
+	}
+
+	totalTokens := sdk.NewDecCoinsFromCoins(total...)
+	// truncated decimals are implicitly added to provider
+	consumerTokens, _ := totalTokens.MulDec(frac).TruncateDecimal()
+	providerTokens := total.Sub(consumerTokens)
+
+	return types.NextFeeDistributionEstimate{
+		CurrentHeight:        ctx.BlockHeight(),
+		LastHeight:           lastH.GetHeight(),
+		NextHeight:           nextH,
+		DistributionFraction: ConsumerRedistributeFrac,
+		Total:                totalTokens.String(),
```

Comment:
Could be more useful to return directly provider rewards and consumer rewards, instead of the total.

Reply:
Wait, that's `Provider` and `Consumer`. Yeah, could we change the names to sth more intuitive, e.g., "ToProvider".

Reply:
Changed.

Reply:
Done.